sdk/go/common/workspace/plugins.go

```diff
@@ -443,36 +443,38 @@ func (source *githubSource) Download(
 	return getHTTPResponse(req)
 }
 
-// pluginURLSource can download a plugin from a given PluginDownloadURL, it doesn't support GetLatestVersion
-type pluginURLSource struct {
-	name              string
-	kind              PluginKind
-	pluginDownloadURL string
+// httpSource can download a plugin from a given http url, it doesn't support GetLatestVersion
+type httpSource struct {
+	name string
+	kind PluginKind
+	url  string
 }
 
-func newPluginURLSource(name string, kind PluginKind, pluginDownloadURL string) *pluginURLSource {
-	return &pluginURLSource{
-		name:              name,
-		kind:              kind,
-		pluginDownloadURL: pluginDownloadURL,
+func newHTTPSource(name string, kind PluginKind, url *url.URL) *httpSource {
+	contract.Requiref(
+		url.Scheme == "http" || url.Scheme == "https",
+		"url", `scheme must be "http" or "https", was %q`, url.Scheme)
```

Comment:
Did this work previously for `file://`? Because then this is a change in behavior.

Note that it's possible to register a custom protocol with net/http. Example at: https://pkg.go.dev/net/http#NewFileTransport

Reply:
No this was only ever using plain http client with nothing registered, and given this runs in the CLI it's not really possible for people to call that to register their own function either.